platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ConfigurationMigrationManager.java

```diff
@@ -231,14 +246,23 @@ private MigrationReport doImport(
                 + productVersion
                 + ConfigurationMigrationManager.EXPORT_EXTENSION);
 
+    final Path keyFile = exportDirectory.resolve(ConfigurationMigrationManager.EXPORT_KEY_NAME);
+
     try {
+      Validate.isTrue(checksumIsValid(exportFile));
       delegateToImportMigrationManager(report, exportFile);
     } catch (MigrationException e) {
       report.record(e);
     } catch (SecurityException e) {
       report.record(new MigrationException(Messages.IMPORT_SECURITY_ERROR, exportFile, e));
     } catch (RuntimeException e) {
       report.record(new MigrationException(Messages.IMPORT_INTERNAL_ERROR, exportFile, e));
+    } catch (NoSuchPaddingException e) {
+      report.record(new MigrationException(Messages.IMPORT_INTERNAL_ERROR, keyFile, e));
+    } catch (NoSuchAlgorithmException e) {
```

Comment:
✏️ This branch identical to the one above it.

Reply:
Collapsed